src/Presentation/ListFacetHtmlBuilder.php

```diff
@@ -102,7 +104,7 @@ private function buildCheckboxesViewModel( FacetConfig $config, PropertyConstrai
 
 		foreach ( $this->getValuesAndCounts( $config ) as $i => $valueCount ) {
 			$checkboxes[] = [
-				'label' => $valueCount->value,
+				'label' => $this->localizedTextLookup->getLabelFromEntityIdString( (string)$valueCount->value ),
```

Comment:
This is wrong. There is no guarantee that the value is an item id. The [logic that looks at the type of the property of the facet that I described for the frontend approach is still needed](https://github.com/ProfessionalWiki/WikibaseFacetedSearch/issues/136#issuecomment-2601041463).